src/ValueObject/DateValueObject.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme\ValueObject;
+
+use Drupal\Component\Datetime\DateTimePlus;
+
+/**
+ * Handle information about a date.
+ */
+class DateValueObject extends ValueObjectBase implements DateValueObjectInterface {
+
+  /**
+   * The day property.
+   *
+   * @var string
+   */
+  private $day;
+
+  /**
+   * The month property.
+   *
+   * @var string
+   */
+  private $month;
+
+  /**
+   * The year property.
+   *
+   * @var string
+   */
+  private $year;
+
+  /**
+   * The weekDay property.
+   *
+   * @var string
+   */
+  private $weekDay;
+
+  /**
+   * The month name property.
+   *
+   * @var string
+   */
+  private $monthName;
+
+  /**
+   * DateValueObject constructor.
+   *
+   * @param string $day
+   *   The date day.
+   * @param string $month
+   *   The date month.
+   * @param string $year
+   *   The date year.
+   * @param string|int|null $weekDay
+   *   The day of the week.
+   */
+  private function __construct(string $day, string $month, string $year, string $weekDay = NULL) {
+    $this->day = $day;
+    $this->month = $month;
+    $this->year = $year;
+    $this->weekDay = $weekDay;
+
+    $date = new DateTimePlus(implode('-', [$year, $month, $day]));
+
+    $this->weekDay = empty($weekDay) ?
+      $date->format('l') :
+      $weekDay;
+    $this->monthName = $date->format('F');
```

Comment:
I am not comfortable to have this `monthName` created only by the `DateTimePlus` and not an override if it's `NULL`

Reply:
Fixed.